src/patches.cpp

```diff
@@ -26,49 +26,60 @@
 #include "entity/ccsplayerpawn.h"
 #include "entity/cbasemodelentity.h"
 #include "addresses.h"
+#include "patches.h"
 
 #include "tier0/memdbgon.h"
 
 CMemPatch g_CommonPatches[] =
 {
-	CMemPatch(&modules::server, sigs::MovementUnlock, sigs::Patch_MovementUnlock, "ServerMovementUnlock"),
-	CMemPatch(&modules::vscript, sigs::VScriptEnable, sigs::Patch_VScriptEnable, "VScriptEnable"),
-	CMemPatch(&modules::server, sigs::CheckJumpButtonWater, sigs::Patch_CheckJumpButton, "FixWaterFloorJump"),
+	CMemPatch("ServerMovementUnlock", "ServerMovementUnlock"),
+	CMemPatch("VScriptEnable", "VScriptEnable"),
+	CMemPatch("CheckJumpButtonWater", "FixWaterFloorJump"),
 };
 
 CMemPatch g_ClientPatches[] =
 {
-	CMemPatch(&modules::client, sigs::MovementUnlock, sigs::Patch_MovementUnlock, "ClientMovementUnlock"),
+	CMemPatch("ClientMovementUnlock", "ClientMovementUnlock"),
 };
 
 #ifdef _WIN32
 CMemPatch g_ToolsPatches[] =
 {
 	// Remove some -nocustomermachine checks without needing -nocustomermachine itself as it can break stuff, mainly to enable device selection in compiles
-	CMemPatch(&modules::hammer, sigs::HammerNoCustomerMachine, sigs::Patch_HammerNoCustomerMachine, "HammerNoCustomerMachine", 4),
+	CMemPatch("HammerNoCustomerMachine", "HammerNoCustomerMachine", 4),
 };
 #endif
 
-void InitPatches()
+bool InitPatches(CGameConfig *g_GameConfig)
 {
 	for (int i = 0; i < sizeof(g_CommonPatches) / sizeof(*g_CommonPatches); i++)
-		g_CommonPatches[i].PerformPatch();
+	{
+		if (!g_CommonPatches[i].PerformPatch(g_GameConfig))
+			return false;
+	}
 
 	// Dedicated servers don't load client
 	if (!CommandLine()->HasParm("-dedicated"))
 	{
 		for (int i = 0; i < sizeof(g_ClientPatches) / sizeof(*g_ClientPatches); i++)
-			g_ClientPatches[i].PerformPatch();
+		{
+			if (!g_ClientPatches[i].PerformPatch(g_GameConfig))
+				return false;
+		}
 	}
 
 #ifdef _WIN32
 	// None of the tools are loaded without, well, -tools
 	if (CommandLine()->HasParm("-tools"))
 	{
 		for (int i = 0; i < sizeof(g_ToolsPatches) / sizeof(*g_ToolsPatches); i++)
-			g_ToolsPatches[i].PerformPatch();
+		{
+			if (!g_ToolsPatches[i].PerformPatch(g_GameConfig))
+				return false;
+		}
 	}
 #endif
+	return true;
```

Comment:
As mentioned earlier, I really don't think we should fail the plugin when any single patch breaks.

Reply:
See https://github.com/Source2ZE/CS2Fixes/pull/36#discussion_r1361057725